2996_reflection/reflection.md

```diff
@@ -7263,17 +7266,64 @@ consteval info naming_class() const;
+  * [#.#]{.pnum} Otherwise, if `$P$` appears at or within a default argument ([dcl.fct.default]) that is used by an invocation of a function ([expr.call]) that appears at point `$Q$`, `$eval-point$($Q$)`.
+  * [#.#]{.pnum} Otherwise, if the immediate scope of `$P$` is a function parameter scope introduced by a declaration `$D$` that is not reachable, a point with the same immediate scope as the locus of `$D$`.
+  * [#.#]{.pnum} Otherwise, if `$P$` appears at or within the `$constraint-expression$` introduced by the trailing `$requires-clause$` of a function declaration `$D$` ([dcl.decl.general]), a point with the same immediate scope as the locus of `$D$`.
+  * [#.#]{.pnum} Otherwise, if `$P$` appears within a `$consteval-block-declaration$` `$D$` ([dcl.pre]), a point whose immediate scope is the scope inhabited by `$D$`.
+  * [#.#]{.pnum} Otherwise, `$P$`.
+
+[#]{.pnum} An invocation of `current` that appears at a program point `$P$` is value-dependent ([temp.dep.contexpr]) if `$eval-point$($P$)` is enclosed by a template parameter scope.
+
+[#]{.pnum} *Returns*: Let `$P$` be the point at which the invocation of `current` appears, and let `$S$` be the innermost scope enclosing `$eval-point$($P$)` that is either a class scope, a namespace scope, or a function parameter scope that corresponds to a function. An `access_context` whose naming class is the null reflection and whose scope is the class, namespace, or function to which `$S$` corresponds.
+
+::: example
+```cpp
+struct A {
+  int a = 0;
+  consteval A(int p) : a(p) {}
+};
+struct B : A {
+  using A::A;
+  consteval B(int p, int q) : A(p * q) {}
+  info s = access_context::current().scope();
+};
+
+struct Agg {
+  consteval bool eq(info rhs = access_context::current().scope()) {
+    return s == rhs;
+  }
+  info s = access_context::current().scope();
+};
 
-* [#.#]{.pnum} If `current()` is a default member initializer ([class.mem.general]), or a subexpression thereof, the point at which the constructor definition or aggregate initialization that uses the default member initializer appears.
-* [#.#]{.pnum} Otherwise, if `current()` is a default argument ([dcl.fct.default]), or a subexpression thereof, the point at which the invocation of the function that uses the default argument ([expr.call]) appears.
-* [#.#]{.pnum} Otherwise, the point at which the invocation of `current()` appears.
+namespace NS {
+static_assert(Agg{}.s == access_context::current().scope());  // OK
+static_assert(Agg{}.eq());  // OK
+static_assert(B(1).s == ^^B);  // OK
+static_assert(is_constructor(B{1, 2}.s) && parent_of(B{1, 2}.s) == ^^B);  // OK
 
-[#]{.pnum} *Returns*: An `access_context` whose naming class is the null reflection and whose scope is defined as follows:
+auto fn() -> [:is_namespace(access_context::current().scope()) ? ^^int : ^^bool:];
+static_assert(^^decltype(fn()) == ^^int);  // OK
```

Comment:
Maybe `type_of(^^fn) == ^^auto()->int`?

Reply:
Yeah. You know how when you write `template <int V> void f()` and then you try to call it as `f<int()>()` and that just doesn't work because it turns out that `int()` is a function type?

Reply:
Whatever, done.